app/models/concerns/foreman/sti.rb

```diff
@@ -15,13 +15,5 @@ def new(attributes = nil, &block)
         super
       end
     end
-
-    def save(*args, **kwargs)
```

Comment:
So this was the whole problem? That it set an instance variable on the class? That looks nasty and I wonder how that worked in the past.

From reading the Rails source I get the impression should handle this itself.